src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -50,7 +50,7 @@ static void send_ack_if_datasharing(
         const SequenceNumber_t& sequence_number)
 {
     // If not datasharing, we are done
-    if (!writer || !writer->is_datasharing_writer())
+    if (!writer || !writer->is_datasharing_writer() || writer->is_on_same_process())
```

Comment:
Small nitpick. Wouldn't it be faster to prevent entry on this function altogether if `writer->is_on_same_process()` is true?

Reply:
77caf2a4b5c308441d44e1e7a48e7351d4e76751